src/phoenix/server/api/routers/trace_handler.py

```diff
@@ -54,7 +60,122 @@ async def post(self, request: Request) -> Response:
         for resource_spans in req.resource_spans:
             project_name = get_project_name(resource_spans.resource.attributes)
             for scope_span in resource_spans.scope_spans:
-                for span in scope_span.spans:
-                    self.traces.put(decode(span), project_name=project_name)
+                for otlp_span in scope_span.spans:
+                    span = decode(otlp_span)
+                    async with self.db() as session:
+                        await _insert_span(session, span, project_name)
+                    self.traces.put(span, project_name=project_name)
                     await asyncio.sleep(0)
         return Response()
+
+
+async def _insert_span(session: AsyncSession, span: Span, project_name: str) -> None:
+    if not (
+        project_rowid := await session.scalar(
+            text("SELECT rowid FROM projects WHERE name = :name;"),
```

Comment:
Is this safe to sql injection?

Reply:
yes. colon is one of the three ways for parameter binding like ?, but sqlalchemy only allows the colon syntax.